lua/ttt2/libraries/fileloader.lua

```diff
@@ -0,0 +1,76 @@
+---
+-- A file loading library for a custom ttt2 file loader
+-- @author Mineotopia
+
+if SERVER then
+	AddCSLuaFile()
+end
+
+local fileFind = file.Find
+local stringRight = string.Right
+
+fileloader = fileloader or {}
+
+---
+-- Sets up files by scanning through directories and including them into the runtime.
+-- @note Has to be run on both server and client for client and shared files.
+-- @param string path The path to search in
+-- @param[default=false] boolean deepsearch If true, subfolders are scanned
```

Comment:
This means it is not fully recursive but only one layer deep right? we should state that explicitly imo

Reply:
yes, it is. I'll update the description